### ISSUE TITLE
[Bug] Latest Windows Apache 2.4.57 download file no permissions

### ISSUE THREAD

**Issue by BBloke** [evocms-community/evolution#310]
### Summary

When using FileDownload plug in to download files, windows apache 2.4.57 returns an error for files with spaces in the name.

### Steps to reproduce.

Try and download a file using the filedownload plugin with spaces in and apache reports no permissions.

example URU: http://test.com/home?d=file with spaces.pdf

### Observed behavior

I changed the following line in .htaccess:

RewriteRule ^(.*)$ index.php?q=$1 [L,QSA]

to

RewriteRule ^(.*)$ index.php?q=$1 "[L,QSA,B= ?,BNP]"

### Expected behavior

File downloading worked as expected after the change.

### EvoCMS version

1.4.24

### Apache/nginx version

Windows Apache 2.4.57

### PHP version

8.1.21

### MySQL version

MariaDB 10.6.12

**Comment by Pathologic** [comment]
What is FileDownload plugin?

**Comment by BBloke** [comment]
FileDownload was a snippet that listed files from a folder using template variables.  The download links are appended parameters to the URL. The plugin that went with it allowed the downloads to work.

The problem happened after updating apache for windows.

Apache error log pointed to rewrite rules.  Whilst the issue isn't Evo or FileDownload the easiest way to fix it was change the rewrite rule in htaccess.

**Comment by Pathologic** [comment]
I think that plugin code is the problem, not htaccess.

**Comment by BBloke** [comment]
I've looked at it more and without the htaccess amendment it's apache that fails not Evo.  This is because it sees spaces in the URL.

The FileDownload snippet creates a url depending on filenames found in the folder.  If the filenames contain spaces then this is shown on the webpage

```php
/page1/test.html?d=file name.pdf
```

When clicking the link the end URL contains %20 as spaces

```php
/page1/test.html?d=file%20name.pdf
```

This fails as apache comes back with: **AH10411: Rewritten query string contains control characters or spaces, referer:**

I've tried encapsulating the parameter in quotes. Didn't help.
